### PR TITLE
PAPP-4148: Adding action list groups and who is on call

### PR DESCRIPTION
**Pull request by ebhoir-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
- [ ] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: xmatters.json [hunk 1]
```diff
@@ -2318,6 +2318,722 @@
...
+                    "example_values": [
+                        1
+                    ]
+                }
+            ],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get oncall user",
+            "description": "Get iinformation about who is on call over a given timeframe",
```
Comment (pdesai-crest) [inline xmatters.json:2645]: typo

**Comment by dgopani-crest** [comment]
Please increment minor app version

File: xmatters.json [hunk 2]
```diff
@@ -2318,6 +2318,722 @@
                 }
             ],
             "versions": "EQ(*)"
+        },
+        {
+            "action": "list groups",
+            "description": "Get information about multiple groups matching a property name/value",
```
Comment (dgopani-crest) [inline xmatters.json:2324]: The description seems incorrect based on inputs params. There is no property name/value input params

**Review by pdesai-crest — APPROVED** [review]
Replace exit with sys.exit

**Comment by dgopani-crest** [comment]
Please add example values for action_result.parameter.members.* datapath

**Comment by dgopani-crest** [comment]
Fix static test job failure for fips_compliant flag

File: xmatters.json [hunk 3]
```diff
@@ -2318,6 +2318,722 @@
...
+                    ]
+                }
+            ],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get oncall user",
+            "description": "Get iinformation about who is on call over a given timeframe",
+            "type": "investigate",
+            "identifier": "who_is_oncall",
```
Comment (ngaglani-crest) [inline xmatters.json:2647]: Can we change the identifier to `get oncall user`?

File: release_notes/unreleased.md [hunk 4]
```diff
@@ -1 +1,2 @@
 **Unreleased**
+* Added new action 'list groups' and 'who is on call'. [PAPP-4148]
```
Comment (ngaglani-crest) [inline release_notes/unreleased.md:2]:
```suggestion
* Added new actions, 'list groups' and 'get oncall user'. [PAPP-4148]
```

File: xmatters_connector.py [hunk 5]
```diff
@@ -553,13 +655,18 @@ def handle_action(self, param):
...
             result = self._initiate_event(param)
+        if action == self.ACTION_ID_LIST_GROUPS:
+            result = self._list_groups(param)
+        if action == self.ACTION_ID_WHO_IS_ONCALL:
+            result = self._who_is_oncall(param)
 
         return result
 
 
 if __name__ == '__main__':
```
Comment (ngaglani-crest) [inline xmatters_connector.py:666]: Replace `exit()` with `sys.exit()` wherever applicable.

**Comment by dgopani-crest** [comment]
We can remove exclude.txt as the app is opensource now

File: xmatters.json (hunk 2 above)
Comment (ebhoir-crest) [reply xmatters.json]: Modified the description.

File: xmatters.json (hunk 1 above)
Comment (ebhoir-crest) [reply xmatters.json]: done

File: release_notes/unreleased.md (hunk 4 above)
Comment (ebhoir-crest) [reply release_notes/unreleased.md]: Modified as suggested

**Comment by dgopani-crest** [comment]
@ebhoir-crest, Is these actions use customer view? no `column_order` assigned in both the action

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**